pkg/monitor/format/format.go

```diff
@@ -122,6 +122,16 @@ func (m *MonitorFormatter) traceEvents(prefix string, data []byte) {
 	}
 }
 
+func (m *MonitorFormatter) traceSockEvents(prefix string, data []byte) {
+	tn := monitor.TraceSockNotify{}
+
+	binary.Read(bytes.NewReader(data), byteorder.Native, &tn)
```

Comment:
Please check for error here and `log.Fatal` otherwise, so we catch if at some point `data` is of wrong length etc.

Reply:
Cilium monitor is for debugging events from the datapath, we don't want to fatal in case of malformed data. For soccket-lb trace events, we will have fixed size trace messages, so I'm not sure if we are likely to have this case "if at some point data is of wrong length". Also, we check alignment of userspace and the corresponding datapath structs.

Reply:
Silently ignoring an error is not great I agree. I think we have previously cut corners here just because we do control the format strongly on both sides and we have these other mechanisms like the alignment checker to ensure it won't fail.

I am generally very cautious about `log.Fatal()`, there are few situations I think it's appropriate due to the concerns @aditighag raises here: We can cause a bigger problem simply by fataling than we would have by just logging some sort of error/warning and moving on.

At a glance, the other code does check the error from `binary.Read()`, so let's just follow that same format here.